app/controllers/api/storage.php

```diff
@@ -24,9 +25,449 @@
+        $maximumFileSize??=$bucket->getAttribute('maximumFileSize', (int)App::getEnv('_APP_STORAGE_LIMIT', 0));
+        $allowedFileExtensions??=$bucket->getAttribute('allowedFileExtensions', []);
+        $enabled??=$bucket->getAttribute('enabled', true);
+        $encryption??=$bucket->getAttribute('encryption', true);
+        $antiVirus??=$bucket->getAttribute('antiVirus', true);
+
+        $bucket = $dbForInternal->updateDocument('buckets', $bucket->getId(), $bucket
+                ->setAttribute('name',$name)
+                ->setAttribute('$read',$read)
+                ->setAttribute('$write',$write)
+                ->setAttribute('maximumFileSize',$maximumFileSize)
+                ->setAttribute('allowedFileExtensions',$allowedFileExtensions)
+                ->setAttribute('enabled',$enabled)
+                ->setAttribute('encryption',$encryption)
+                ->setAttribute('antiVirus',$antiVirus)
+        );
+
+        $audits
+            ->setParam('event', 'storage.buckets.update')
+            ->setParam('resource', 'storage/buckets/' . $bucket->getId())
+            ->setParam('data', $bucket->getArrayCopy())
+        ;
+
+        $usage->setParam('storage.buckets.update', 1);
+
+        $response->dynamic($bucket, Response::MODEL_BUCKET);
+    });
+
+App::delete('/v1/storage/buckets/:bucketId')
+    ->desc('Delete Bucket')
+    ->groups(['api', 'storage'])
+    ->label('scope', 'buckets.write')
+    ->label('event', 'storage.buckets.delete')
+    ->label('sdk.auth', [APP_AUTH_TYPE_KEY])
+    ->label('sdk.namespace', 'storage')
+    ->label('sdk.method', 'deleteBucket')
+    ->label('sdk.description', '/docs/references/storage/delete-bucket.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_NOCONTENT)
+    ->label('sdk.response.model', Response::MODEL_NONE)
+    ->param('bucketId', '', new UID(), 'Bucket unique ID.')
+    ->inject('response')
+    ->inject('dbForInternal')
+    ->inject('audits')
+    ->inject('deletes')
+    ->inject('events')
+    ->inject('usage')
+    ->action(function ($bucketId, $response, $dbForInternal, $audits, $deletes, $events, $usage) {
+        /** @var Appwrite\Utopia\Response $response */
+        /** @var Utopia\Database\Database $dbForInternal */
+        /** @var Appwrite\Event\Event $audits */
+        /** @var Appwrite\Event\Event $deletes */
+        /** @var Appwrite\Event\Event $events */
+        /** @var Appwrite\Stats\Stats $usage */
+
+        $bucket = $dbForInternal->getDocument('buckets', $bucketId);
+
+        if ($bucket->isEmpty()) {
+            throw new Exception('Bucket not found', 404);
+        }
+
+        $deletes
+            ->setParam('type', DELETE_TYPE_DOCUMENT)
+            ->setParam('document', $bucket)
+        ;
+
+        if(!$dbForInternal->deleteDocument('buckets', $bucketId)) {
+            throw new Exception('Failed to remove project from DB', 500);
+        }
```

Comment:
We should set the event for the deletes worker after a successful deletion of the bucket. Otherwise we might not remove the bucket document - but trigger the delete worker 👍🏻